packages/apollo-server-plugin-base/src/index.ts

```diff
@@ -65,6 +67,9 @@ export interface ApolloServerPlugin<
 }
 
 export interface GraphQLServerListener {
+  schemaDidChange?(
+    schemaContext: GraphQLSchemaContext,
+  ): void;
```

Comment:
Would it make the correctness a lot harder to make this method async (ideally just plain Promise<void> async as opposed to the weirder ValueOrPromise)? Making all plugin methods async is in the air for AS3 (#4999). I can imagine it might make the logic harder to implement correctly.

Reply:
(Paraphrasing from a Slack convo we had with Trevor) We've decided to keep this sync, as there's concerns about listeners making expensive async calls stalling execution. If users want to do async things in response to a schema load/update, they can queue the changes and manage async queue consumption themselves.